clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -94,7 +94,7 @@ private Utils() {}
 
     // This matches URIs of formats: host:port and protocol://host:port
     // IPv6 is supported with [ip] pattern
-    private static final Pattern HOST_PORT_PATTERN = Pattern.compile("^(?:[a-zA-Z][a-zA-Z\\d+-.]*://)?\\[?([0-9a-zA-Z\\-._%:]+)\\]?:([0-9]+)$");
+    private static final Pattern HOST_PORT_PATTERN = Pattern.compile("^(?:[a-zA-Z][a-zA-Z\\d+_.-]*://)?\\[?([0-9a-zA-Z\\-._%:]+)]?:([0-9]+)$");
```

Comment:
Also, the removal of the final `\\` is surprising. Can you please confirm this was intentional?

Reply:
Yes, I delete it because I have receive an alert from my idea, it said that this escape is redundant.
I have test it in [regex101](https://regex101.com/), I think it work as expected.